mesonbuild/backend/ninjabackend.py

```diff
@@ -612,6 +612,14 @@ def generate(self):
             self.build_elements = []
             self.generate_phony()
             self.add_build_comment(NinjaComment('Build rules for targets'))
+
+            # Optionally capture compile args per buildtype and target, for later use (i.e. VisStudio project's NMake intellisense include dirs, defines, and compile options).
+            if captured_compile_args_per_buildtype_and_target is not None:
+                for target in self.build.get_targets().values():
+                    if isinstance(target, build.BuildTarget):
+                        buildtype = target.get_option(OptionKey('buildtype'))
+                        captured_compile_args_per_buildtype_and_target[buildtype][target.get_id()] = self.generate_common_compile_args_per_src_type(target)
+
```

Comment:
FWICT `captured_compile_args_per_buildtype_and_target` is only written meaning it is actually an out parameter. If that is the case then that needs to be changed as those are not allowed. The function should store this in a local variable and return that as a return value instead.

Reply:
Thanks Jussi.  I think this is now addressed in [e816dd7](https://github.com/mesonbuild/meson/pull/11049/commits/e816dd79d2ea21e1fbdfe6cf5210a12fc437eb76).

I can totally understand and if it's a rule then it's a rule and I'm fine with this suggestion.  I just wanted to add though, that the ability to use an 'out' param can arguably make for cleaner or more concise code.  E.g. Passing in a valid arg acts as both an implicit bool argument to the called function that you want to fill the dict with captured compile values (or whatever) as well as providing the container to be filled in.

Additionally, in this particular case, we have the behaviour of -
- only ninja backend 'generate's are expected to potentially produce captured compile options, if asked.
- and only vs backend 'generate's are expected to make use of previously gathered captured compile options.

so that behaviour could be conveyed entirely through passing None or a valid dict to the ninja backend's 'generate' to fill in, and then passing a valid, filled in 'capture dict' to the vs backend's 'generate', through the same arg, to read from... which is _arguably_ a bit simpler and/or cleaner;  the alternative now being one bool arg to request a capture (which only needs to be heeded in the ninja backend) through returned value (only used in ninja backends) as well as an additional 'in'-only arg that's only used in vs backends.

Anyway, like I said, I'm happy to make the change, but just wanted to mention that I think I actually prefer the original, multi-functional arg style 😄

Reply:
I see at the bottom of this whole PR, it currently says, '1 change requested'.  I'm just wondering if there's some automatic github PR workflow gating that I'm not aware of, that means I need to do or tick something, in addition to my reply immediately above, in order to notify reviewers to get issues and requested changes resolved?  I can obviously see the 'resolve conversation' at the bottom of this issue/thread but that's surely not for me to do;  I shouldn't be able to decide whether I have resolved an issue raised by someone else.

I'm not impatient (I know a good code review on changes this size is usually time consuming);  just wondering if I've missed something that others might be waiting on.